Problem builing on ARM64

### socketcan version

4.2.2

### Node.js version

22

### Linux distribution and kernel

Ubuntu

### CAN interface type

Both

### What happened?

Update to new version on node red gives

2026-06-26T12:42:59.931Z [err] npm error code 1
2026-06-26T12:42:59.933Z [err] npm error path /data/node_modules/socketcan
2026-06-26T12:42:59.934Z [err] npm error command failed
2026-06-26T12:42:59.936Z [err] npm error command sh -c node-gyp rebuild
2026-06-26T12:42:59.939Z [err] npm error make: Entering directory '/data/node_modules/socketcan/build'
2026-06-26T12:42:59.939Z [err] npm error   CXX(target) Release/obj.target/can/native/can.o
2026-06-26T12:42:59.939Z [err] npm error   SOLINK_MODULE(target) Release/obj.target/can.node
2026-06-26T12:42:59.939Z [err] npm error   COPY Release/can.node
2026-06-26T12:42:59.939Z [err] npm error   CXX(target) Release/obj.target/can_signals/native/signals.o
2026-06-26T12:42:59.939Z [err] npm error make: Leaving directory '/data/node_modules/socketcan/build'
2026-06-26T12:43:00.009Z [err] npm error gyp info it worked if it ends with ok
2026-06-26T12:43:00.009Z [err] npm error gyp info using node-gyp@13.0.0
2026-06-26T12:43:00.009Z [err] npm error gyp info using node@24.16.0 | linux | arm64
2026-06-26T12:43:00.009Z [err] npm error gyp info find Python using Python version 3.12.13 found at "/usr/bin/python3"
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn /usr/bin/python3
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args [
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '/data/node_modules/node-gyp/gyp/gyp_main.py',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args 'binding.gyp',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-f',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args 'make',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-I',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '/data/node_modules/socketcan/build/config.gypi',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-I',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '/data/node_modules/node-gyp/addon.gypi',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-I',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '/root/.cache/node-gyp/24.16.0/include/node/common.gypi',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dlibrary=shared_library',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dvisibility=default',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dnode_root_dir=/root/.cache/node-gyp/24.16.0',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dnode_gyp_dir=/data/node_modules/node-gyp',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dnode_lib_file=/root/.cache/node-gyp/24.16.0/<(target_arch)/node.lib',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dmodule_root_dir=/data/node_modules/socketcan',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Dnode_engine=v8',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '--depth=.',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '--no-parallel',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '--generator-output',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args 'build',
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args '-Goutput_dir=.'
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args ]
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn make
2026-06-26T12:43:00.009Z [err] npm error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
2026-06-26T12:43:00.009Z [err] npm error In file included from ../native/[can.cc:39](http://can.cc:39/):
2026-06-26T12:43:00.009Z [err] npm error /usr/include/sys/poll.h:1:2: warning: #warning redirecting incorrect #include <sys/poll.h> to <poll.h> [-Wcpp]
2026-06-26T12:43:00.009Z [err] npm error     1 | #warning redirecting incorrect #include <sys/poll.h> to <poll.h>
2026-06-26T12:43:00.009Z [err] npm error       |  ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc](http://signals.cc/): In function 'uint64_t _getvalue(const uint8_t*, uint32_t, uint32_t, ENDIANESS)':
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc:75](http://signals.cc:75/):52: error: 'le64toh' was not declared in this scope
2026-06-26T12:43:00.009Z [err] npm error    75 |     uint64_t d = (byteOrder == ENDIANESS::INTEL) ? le64toh(d_raw) : be64toh(d_raw);
2026-06-26T12:43:00.009Z [err] npm error       |                                                    ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc:75](http://signals.cc:75/):69: error: 'be64toh' was not declared in this scope
2026-06-26T12:43:00.009Z [err] npm error    75 |     uint64_t d = (byteOrder == ENDIANESS::INTEL) ? le64toh(d_raw) : be64toh(d_raw);
2026-06-26T12:43:00.009Z [err] npm error       |                                                                     ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc](http://signals.cc/): In function 'void _setvalue(uint32_t, uint32_t, ENDIANESS, uint8_t*, uint64_t)':
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc:179](http://signals.cc:179/):52: error: 'le64toh' was not declared in this scope
2026-06-26T12:43:00.009Z [err] npm error   179 |     uint64_t o = (endianess == ENDIANESS::INTEL) ? le64toh(o_raw) : be64toh(o_raw);
2026-06-26T12:43:00.009Z [err] npm error       |                                                    ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc:179](http://signals.cc:179/):69: error: 'be64toh' was not declared in this scope
2026-06-26T12:43:00.009Z [err] npm error   179 |     uint64_t o = (endianess == ENDIANESS::INTEL) ? le64toh(o_raw) : be64toh(o_raw);
2026-06-26T12:43:00.009Z [err] npm error       |                                                                     ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc:187](http://signals.cc:187/):43: error: 'htole64' was not declared in this scope
2026-06-26T12:43:00.009Z [err] npm error   187 |     o = (endianess == ENDIANESS::INTEL) ? htole64(o) : htobe64(o);
2026-06-26T12:43:00.009Z [err] npm error       |                                           ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error ../native/[signals.cc:187](http://signals.cc:187/):56: error: 'htobe64' was not declared in this scope
2026-06-26T12:43:00.009Z [err] npm error   187 |     o = (endianess == ENDIANESS::INTEL) ? htole64(o) : htobe64(o);
2026-06-26T12:43:00.009Z [err] npm error       |                                                        ^~~~~~~
2026-06-26T12:43:00.009Z [err] npm error make: *** [can_signals.target.mk:112: Release/obj.target/can_signals/native/signals.o] Error 1
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! build error
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! stack Error: make failed with exit code: 2
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! stack at ChildProcess.<anonymous> (/data/node_modules/node-gyp/lib/build.js:219:23)
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! System Linux 6.6.52-rt43-g6565678da008-dirty
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! command "/usr/local/bin/node" "/data/node_modules/.bin/node-gyp" "rebuild"
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! cwd /data/node_modules/socketcan
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! node -v v24.16.0
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! node-gyp -v v13.0.0
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! $npm_package_name socketcan
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! $npm_package_version 4.2.2
2026-06-26T12:43:00.009Z [err] npm error gyp ERR! not ok
2026-06-26T12:43:00.024Z [err] npm error A complete log of this run can be found in: /data/.npm/_logs/2026-06-26T12_17_27_522Z-debug-0.log
2026-06-26T12:43:00.075Z rc=1

Probable cause 
"endian.h"  missing in  native/signals.cc



### What did you expect to happen?

A clean build

### Minimal reproduction

```javascript
ARM64
Node 22
Update from any version to 4.2.x
```

--- a/.github/workflows/cicd.yml
+++ b/.github/workflows/cicd.yml
@@ -8,11 +8,13 @@ permissions:
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
 
     strategy:
+      fail-fast: false
       matrix:
         node-version: [ 22.x, 24.x ]
+        os: [ ubuntu-latest, ubuntu-24.04-arm ]
 
     steps:
       - name: Checkout

--- a/native/signals.cc
+++ b/native/signals.cc
@@ -15,6 +15,8 @@
  */
 #include <napi.h>
 
+#include <endian.h>   // le64toh/be64toh/htole64/htobe64 (not pulled in transitively on all arches)
+
 #include <algorithm>
 #include <bit>
 #include <cstdint>

--- a/package.json
+++ b/package.json
@@ -5,7 +5,7 @@
     "email": "sebastian@sebastianhaas.info"
   },
   "description": "A SocketCAN abstraction layer for NodeJS.",
-  "version": "4.2.2",
+  "version": "4.2.3",
   "license": "MIT",
   "keywords": [
     "can",

